docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -126,3 +126,19 @@ A Workflow Execution Timeout can be used to limit the duration of Workflow Execu
 You can use the [Continue-As-New](/concepts/what-is-continue-as-new) feature to close the current Workflow Execution and create a new Workflow Execution in a single atomic operation.
 The Workflow Execution spawned from Continue-As-New has the same Workflow Id, a new Run Id, and a fresh Event History and is passed all the appropriate parameters.
 For example, it may be reasonable to use Continue-As-New once per day for a long-running Workflow Execution that is generating a large Event History.
+
+### Limits
```

Comment:
We should provide the users with more information on what these limits do and why. Something like this:

> When executing Commands, Temporal may consume resources that it only releases later. For example, the ScheduleActivityTask Command adds an entry to the Workflow's pending Activities. Temporal removes this entry only when the activity terminates. As a result, having too many pending Activities may impede progress on the Workflow that created them.
> 
> To avoid this, whenever Temporal executes a Workflow Task that contains a ScheduleActivityTask Command, which would cause the Workflow to exceed its limit of pending activities, Temporal fails that Workflow Task Execution. It then adds a WorkflowTaskFailed event to the Workflow's history, containing a cause of WORKFLOW_TASK_FAILED_CAUSE_PENDING_ACTIVITIES_LIMIT_EXCEEDED within its failure field.
> 
> Temporal retries Workflow Tasks, so if a sufficient number of pending activities terminate, the Workflow Task to schedule more Activities may eventually succeed. However, suppose the Workflow Task contains Commands that exceed this limit even if there are no pending Activities when the Command executes. In that case, the Workflow Task will eventually fail.
> 
> Temporal enforces this constraint similarly for the SignalExternalWorkflowExecution, RequestCancelExternalWorkflowExecution, and StartChildWorkflowExecution Commands. For each Workflow, the number of pending Activities, outgoing Signals, cancel requests to external Workflows, and Child Workflows that it may contain are each limited to 50,000. For Temporal Cloud customers, this limit is 2,000.

Reply:
Your edits look great. Much simpler way of phrasing it, thank you!